File: backend/loja-api/src/main/java/com/accenture/loja/endereco/model/Endereco.java
package com.accenture.loja.endereco.model;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "enderecos")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Endereco {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String cep;

    private String rua;

    @Column(nullable = false)
    private String numero;

    private String complemento;

    private String bairro;

    private String cidade;

    private String uf;
}

File: frontend/loja-web/src/types/Endereco.ts
export interface Endereco {
  id?: number;
  cep: string;
  numero: string;
  complemento?: string;
  rua?: string;
  logradouro?: string;
  bairro?: string;
  cidade?: string;
  localidade?: string;
  uf?: string;
}

export interface ViaCepResponse {
  cep?: string;
  logradouro?: string;
  localidade?: string;
  rua?: string;
  cidade?: string;
  bairro?: string;
  uf?: string;
  erro?: boolean;
}